### ISSUE TITLE
When zooming a triangle is rendered instead of a rectangle.

### ISSUE THREAD

**Issue by andnovar** [andnovar/kivy#72]

**Comment by andnovar** [comment]
Mpl is sending the number of vertex that are at least visible, for instance if the rectangle drawn is inside the bar it will send 0 paths since no paths are enclosed in that area. In the case of enclosing the right line it tries to create a mesh with the origin and the line enclosed. In the case of enclosing the left line since the origin and the line have the same y is just shown the line. This happens as well in Gtk and in none of the Agg backends neither backend_kivyagg. There is some clipping problem here.

**Comment by matham** [comment]
Ok, but is this a mpl bug then? If so you need to open a issue with them.

**Comment by andnovar** [comment]
![screenshot from 2015-08-21 12 51 27](https://cloud.githubusercontent.com/assets/400600/9416779/bba8bc22-4815-11e5-826c-29fc91970716.png)
![screenshot from 2015-08-21 12 51 35](https://cloud.githubusercontent.com/assets/400600/9416780/bbb6c56a-4815-11e5-8dda-b73840cda18f.png)

**Comment by tacaswell** [comment]
Did this get converted to an mpl issue?  Does this happen on master?  I think there was some work on clipping (which seems to be a persistent problem).

**Comment by andnovar** [comment]
Yes that happens on master. I just opened this issue in matplotlib. The clipping area should be bigger to at least enclose the polygon that is being rendered. Or send the intersections with the coordinates as vertices as well in a way that a new polygon is created inside the visible area.

**Comment by tacaswell** [comment]
I don't think we compute the intersections.  I also thought that there was a cut-out to not clip on filled paths.

**Comment by andnovar** [comment]
Can you extend more on the cut-out to not clip on filled paths ?

**Comment by tacaswell** [comment]
Have a look at https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backends/backend_svg.py#L559  

I miss-remembered where the logic was, I thought it was in the artists, but it is in the backends.  The backends should be getting the full path from the `Rectangle` artist + the affine transforms into `draw_path`.

**Comment by andnovar** [comment]
I'll take a look at that and I'll let you know thanks !
